NDK r26 has a `fake` clang version. It doesn't has support for CWG2518.

--- a/v8/src/handles/handles.h
+++ b/v8/src/handles/handles.h
@@ -129,7 +129,7 @@
     } else {
       // `static_assert(false)` in this else clause was an unconditional error
       // before CWG2518. See https://reviews.llvm.org/D144285
-#if defined(__clang__) && __clang_major__ >= 17
+#if defined(__clang__) && __clang_major__ >= 17 && !defined(__ANDROID__)
       // For non-HeapObjects, there's no on-heap object to dereference, so
       // disallow using operator->.
       //
